pkg/registry/generic/storage_factory.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generic
+
+import (
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/storage"
+)
+
+// StorageFactory is a function signature for producing
+// a storage.Interface from given parameters.
+type StorageFactory func(
```

Comment:
Thanks-- sorry about the run-around, I think the "factory" moniker was my
idea originally, not really sure what I was thinking. :(

On Wed, Nov 11, 2015 at 10:29 AM, Wojciech Tyczynski <
notifications@github.com> wrote:

> In pkg/registry/generic/storage_factory.go
> <https://github.com/kubernetes/kubernetes/pull/16855#discussion_r44566958>
> :
>
> > +distributed under the License is distributed on an "AS IS" BASIS,
> > +WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> > +See the License for the specific language governing permissions and
> > +limitations under the License.
> > +*/
> > +
> > +package generic
> > +
> > +import (
> > +	"k8s.io/kubernetes/pkg/runtime"
> > +	"k8s.io/kubernetes/pkg/storage"
> > +)
> > +
> > +// StorageFactory is a function signature for producing
> > +// a storage.Interface from given parameters.
> > +type StorageFactory func(
>
> Yes - it took storage.Interface before too.
> I will change that to StorageDecorator.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/16855/files#r44566958>.
>